modules/miscutil/lib/upgrades/invenio_2015_01_13_hide_holdings.py

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2014 CERN.
+#
+# Invenio is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Invenio is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the Free Software Foundation,
+# Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+"""Upgrade for explicitly hiding the 'holdings' tab for all collections\
+ except 'Books' without an existing rule"""
+
+from invenio.dbquery import run_sql
+
+depends_on = ['invenio_release_1_1_0']
+
+
+def info():
+    """Upgrade recipe information."""
+    return "Updates the collectiondetailedrecordpagetabs to hide 'holdings' for every collection \
+            except 'Books' without an existing rule."
+
+
+def do_upgrade():
```

Comment:
> while with my amendment, you would take existing collection 1 settings, and apply them to all daugher collections that do not have anything defined. 

This is equal to checking "apply for subcollection", which is not necessarily what you want. Imagine the following scenario:
Collection 1 is the parent of collections 2 and 3, and it has a `collectiondetailedrecordpagetabs` entry to show only usage, comments and metadata. Collection 2 is set to show usage, comments, metadata and references, while collection 3 has no entry in `collectiondetailedrecordpagetabs`. The result is that collection 3 displays all tabs, even though its parent is restricted to only three.

However, if one was to apply the upgrade you propose, collection 3 would suddenly also be restricted by its parents settings, changing the behavior (and if it's one thing this upgrade script should do, it's to preserve behavior).